gui/src/main/java/org/verapdf/gui/ValidateWorker.java

```diff
@@ -64,149 +43,55 @@
-                    GUIConstants.ERROR_IN_VALIDATING, e);
-        }
-        return null;
+    protected Void doInBackground() {
+		try {
+			this.xmlReport = File.createTempFile("veraPDF-tempXMLReport", ".xml");
+			this.xmlReport.deleteOnExit();
+			this.htmlReport = null;
+		} catch (IOException e) {
+			LOGGER.error("Can't create temporary file for XML report", e);
+		}
+		try (InputStream toProcess = new FileInputStream(pdf);
+		OutputStream mrrReport = new FileOutputStream(this.xmlReport)) {
+			Processor processor = new ProcessorImpl();
+			processingResult = processor.validate(toProcess, ItemDetails.fromFile(pdf),
+					settings, mrrReport);
+		} catch (IOException e) {
+			LOGGER.error("Can't open stream from PDF file or can't open stream to temporary XML report file", e);
+		}
+
+        writeHtmlReport();
+
+		return null;
     }
 
     @Override
     protected void done() {
-        this.parent.validationEnded(this.xmlReport, this.htmlReport);
+        this.parent.validationEnded(this.xmlReport, this.htmlReport,
+				this.processingResult);
     }
 
-    private void writeReports(ValidationResult result,
-            MetadataFixerResult fixerResult, FeaturesCollection collection) {
+    private void writeHtmlReport() {
         try {
-            this.xmlReport = File.createTempFile("veraPDF-tempXMLReport",
-                    ".xml");
-            this.xmlReport.deleteOnExit();
-            MachineReadableReport report = MachineReadableReport.fromValues(this.pdf,
-                    this.profile, result, this.settings.isShowPassedRules(),
-                    this.settings.getMaxNumberOfDisplayedFailedChecks(), fixerResult, collection,
-                    this.endTimeOfValidation - this.startTimeOfValidation);
-            try (OutputStream xmlReportOs = new FileOutputStream(this.xmlReport)) {
-                MachineReadableReport.toXml(report, xmlReportOs, Boolean.TRUE);
-            }
-            if (result != null) {
-                this.htmlReport = File.createTempFile("veraPDF-tempHTMLReport",
-                        ".html");
-                this.htmlReport.deleteOnExit();
-                try (InputStream xmlStream = new FileInputStream(this.xmlReport);
-                        OutputStream htmlStream = new FileOutputStream(
-                                this.htmlReport)) {
-                    HTMLReport.writeHTMLReport(xmlStream, htmlStream, this.settings.getProfileWikiPath());
-
-                } catch (IOException | TransformerException e) {
-                    JOptionPane.showMessageDialog(this.parent,
-                            GUIConstants.ERROR_IN_SAVING_HTML_REPORT,
-                            GUIConstants.ERROR, JOptionPane.ERROR_MESSAGE);
-                    LOGGER.error("Exception saving the HTML report", e);
-                    this.htmlReport = null;
-                }
-            }
+			this.htmlReport = File.createTempFile("veraPDF-tempHTMLReport",
```

Comment:
if we did not write a xml report, then we should not create even a temp file for html report